nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -86,12 +83,13 @@ class BtServiceNode : public BT::CoroActionNode
 
     rclcpp::executor::FutureReturnCode rc;
     rc = rclcpp::spin_until_future_complete(node_,
-        future_result, node_loop_timeout_);
-    if (rc != rclcpp::executor::FutureReturnCode::SUCCESS) {
-      return BT::NodeStatus::FAILURE;
-    } else {
+        future_result, std::chrono::milliseconds(service_timeout_));
+    if (rc == rclcpp::executor::FutureReturnCode::SUCCESS) {
       return BT::NodeStatus::SUCCESS;
+    } else if (rc == rclcpp::executor::FutureReturnCode::TIMEOUT) {
```

Comment:
Please add a `RCLCPP_WARN(node_->get_logger(), "Node timed out while executing service call to %s.", service_name_.c_str())'`